protocol/policy.go

```diff
@@ -9,34 +9,38 @@ import (
 type Timestamp uint64
 
 type Policies struct {
-	LibVersion    string
-	HashID        string
-	vrfPrivateKey vrf.PrivateKey
 	VrfPublicKey  vrf.PublicKey
 	EpochDeadline Timestamp
+	LibVersion    string
+	HashID        string
 }
 
-func NewPolicies(epDeadline Timestamp, vrfPrivKey vrf.PrivateKey) *Policies {
-	vrfPublicKey, ok := vrfPrivKey.Public()
-	if !ok {
-		panic(vrf.ErrorGetPubKey)
-	}
+func NewPolicies(epDeadline Timestamp, vrfPublicKey vrf.PublicKey) *Policies {
 	return &Policies{
-		LibVersion:    Version,
-		HashID:        crypto.HashID,
-		vrfPrivateKey: vrfPrivKey,
 		VrfPublicKey:  vrfPublicKey,
 		EpochDeadline: epDeadline,
+		LibVersion:    Version,
+		HashID:        crypto.HashID,
 	}
 }
 
 // Serialize encodes the policy to a byte array with the following format:
 // [lib version, cryptographic algorithm in use, epoch deadline, vrf public key]
 func (p *Policies) Serialize() []byte {
 	var bs []byte
+	bs = append(bs, p.VrfPublicKey...)                             // vrf public key
+	bs = append(bs, util.ULongToBytes(uint64(p.EpochDeadline))...) // epoch deadline
 	bs = append(bs, []byte(p.LibVersion)...)                       // lib Version
 	bs = append(bs, []byte(p.HashID)...)                           // cryptographic algorithms in use
-	bs = append(bs, util.ULongToBytes(uint64(p.EpochDeadline))...) // epoch deadline
-	bs = append(bs, p.VrfPublicKey...)                             // vrf public key
 	return bs
 }
+
+// FIXME: I'm just being lazy :(
+func ParsePubKey(policies []byte) vrf.PublicKey {
+	return vrf.PublicKey(policies[0:vrf.PublicKeySize])
```

Comment:
> it makes the PAD more general, perhaps to be reused in other contexts. 

Sure, this is a good point.

> A thought is that `AssociatedData` could be an interface with serialize/parse methods. What our code ignores right now is that the req/resp will have to be encoded/decoded for the wire anyways, and this conversation would only happen then.
>
> Maybe it can help us somehow in #6.

Yeah, this sounds like a good solution. I think it could definitely help with #6.

> I'm thinking if we could use `unsafe` here to convert between the struct and byte array (like `memcpy` in C), so we would have something like `ParsePolicies(buf []byte) *Policies`. 

You mean because `unsafe.Pointer` is like `void *` in C? I suppose we could use `unsafe` for the conversions, but I'm not sure what the advantage would be over using the scheme (1 byte for len + value).